CIS 2.7.0 Observed a panic: "invalid memory address or nil pointer dereference"

#### Setup Details
 CIS Version : 2.7.0      
 Build: f5networks/k8s-bigip-ctlr:2.7.0      
 BIGIP Version: Big IP 14.1.4.4      
 AS3 Version: 3.30      
 Agent Mode: AS3      
 Orchestration: K8S     
 Orchestration Version:  1.22.5
 Pool Mode: Cluster     
 Additional Setup details: <Platform/CNI Plugins/ cluster nodes/ etc>

#### Description  
After upgrading to 2.7.0 CIS get the following error and crashed:
`E0107 10:18:02.704966       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 186 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1958800, 0x28d9c30)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x1958800, 0x28d9c30)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager.(*CRManager).processService(0xc000246500, 0xc000902be0, 0xc0005b6720, 0xc0003b2200, 0x1, 0x1)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager/worker.go:1898 +0x42a
github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager.(*CRManager).processResource(0xc000246500, 0x1b9c300)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager/worker.go:220 +0x103b
github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager.(*CRManager).customResourceWorker(0xc000246500)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager/worker.go:51 +0x78
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000a6b260)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000a6b260, 0x1d71880, 0xc000a6c2d0, 0x1, 0xc000a02d20)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000a6b260, 0x3b9aca00, 0x0, 0x1c4bd01, 0xc000a02d20)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000a6b260, 0x3b9aca00, 0xc000a02d20)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager.(*CRManager).Start
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager/crManager.go:328 +0x22d
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x174b66a]

goroutine 186 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x1958800, 0x28d9c30)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager.(*CRManager).processService(0xc000246500, 0xc000902be0, 0xc0005b6720, 0xc0003b2200, 0x1, 0x1)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager/worker.go:1898 +0x42a
github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager.(*CRManager).processResource(0xc000246500, 0x1b9c300)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager/worker.go:220 +0x103b
github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager.(*CRManager).customResourceWorker(0xc000246500)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager/worker.go:51 +0x78
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000a6b260)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000a6b260, 0x1d71880, 0xc000a6c2d0, 0x1, 0xc000a02d20)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000a6b260, 0x3b9aca00, 0x0, 0x1c4bd01, 0xc000a02d20)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000a6b260, 0x3b9aca00, 0xc000a02d20)
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager.(*CRManager).Start
	/go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/crmanager/crManager.go:328 +0x22d`


#### Steps To Reproduce
Upgrade to 2.7.0 from 2.6.1 using Helm
I had to manually add the new CRDs for policies and externaldnses as well as update the ClusterRole with the new CRD permissions. The controller doesn't seem to start unless all CRDs are in place and readable, after that is fixed the controller crashes with the error above.

Even a clean install using helm results in old CRDs and old clusterrole permissions.

#### Expected Result 


#### Actual Result  


#### Diagnostic Information
```buildoutcfg
<Configuration files, error messages, logs>
Note: Sanitize the data. For example, be mindful of IPs, ports, application names and URLs
Note: The following F5 article outlines the information required when opening an issue.
https://support.f5.com/csp/article/K60974137
```  

#### Observations (if any)

--- a/docs/RELEASE-NOTES.rst
+++ b/docs/RELEASE-NOTES.rst
@@ -11,6 +11,7 @@ Enhancements
 * :issues:`2051` Fix AS3 Postdelay issue when error occurs
 * :issues:`1918` ExternalDNS adds both VSs to a Wide IP pool
 * :issues:`1872` Support protocol UDP in Services of type LoadBalancer
+* :issues:`2172` Fix Endpoint NodeName validation issue
 
 
 2.7.0

--- a/pkg/crmanager/worker.go
+++ b/pkg/crmanager/worker.go
@@ -1902,7 +1902,7 @@ func (crMgr *CRManager) processService(
 			var members []PoolMember
 			for _, addr := range subset.Addresses {
 				// Checking for headless services
-				if containsNode(nodes, *addr.NodeName) || svc.Spec.ClusterIP == "None" {
+				if svc.Spec.ClusterIP == "None" || (addr.NodeName != nil && containsNode(nodes, *addr.NodeName)) {
 					member := PoolMember{
 						Address: addr.IP,
 						Port:    p.Port,

